terra_notebook_utils/drs.py

```diff
@@ -17,48 +18,77 @@ def _parse_gs_url(gs_url):
     else:
         raise RuntimeError(f'Invalid gs url schema.  {gs_url} does not start with {_GS_SCHEMA}')
 
-def fetch_drs_info(drs_url):
+def enable_requester_pays(env: str=ENV, project: str=WORKSPACE_GOOGLE_PROJECT, workspace: str=WORKSPACE_NAME):
+    global HAS_REQUESTER_PAYS_INIT
+    if HAS_REQUESTER_PAYS_INIT:
+        return
+    HAS_REQUESTER_PAYS_INIT = True
+
+    import urllib.parse
+    encoded_workspace = urllib.parse.quote(workspace)
+    rawls_url = f"https://rawls.dsde-{env}.broadinstitute.org/api/workspaces/{project}/{encoded_workspace}/enableRequesterPaysForLinkedServiceAccounts" # noqa
+    print("Enabling requester pays for your workspace. This will only take a few seconds...")
+    access_token = gs.get_access_token()
+
+    headers = {
+        'authorization': f"Bearer {access_token}",
+        'content-type': "application/json"
+    }
+    resp = requests.put(rawls_url, headers=headers)
+
+    if resp.status_code != 204:
+        print(f"Failed to init requester pays for workspace {project}/{workspace} in env {env}.")
+        print("You will not be able to access drs urls that interact with requester pays buckets.")
+
+def fetch_drs_info(drs_url, env: str):
     """
     Request DRS infromation from martha.
     """
     access_token = gs.get_access_token()
-    martha_url = "https://us-central1-broad-dsde-prod.cloudfunctions.net/martha_v2"
+
+    enable_requester_pays()
+
+    martha_url = f"https://us-central1-broad-dsde-{env}.cloudfunctions.net/martha_v2"
     headers = {
         'authorization': f"Bearer {access_token}",
         'content-type': "application/json"
     }
+
     resp = requests.post(martha_url, headers=headers, data=json.dumps(dict(url=drs_url)))
+
     if 200 == resp.status_code:
         resp_data = resp.json()
     else:
         print(resp.content)
         raise Exception(f"expected status 200, got {resp.status_code}")
     return resp_data
 
-def resolve_drs_for_gs_storage(drs_url):
+def resolve_drs_for_gs_storage(drs_url, env: str=ENV):
```

Comment:
good catch. I originally had it here for testing requester pays logic without being locked into 1 env for all tests, but it proved too complicated because it requires a specific credentials to test out. I made a follow-up ticket to test it out closer to the systems that support the functionality.